### ISSUE TITLE
Workload Identity: User Assigned Identity needs Cluster Admin permissions

### ISSUE THREAD

**Issue by jrauschenbusch** [Azure/kafka-sink-azure-kusto#126]
Switching from Pod Identity to Workload Identity, the assigned UAI now needs Cluster-wide Admin permission, otherwise the following error appears:
```
{
    "exception":{
        "stacktrace": "com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException: Error refreshing IngestionResources. Principal 'aadapp=<redacted>;<redacted>' is not authorized to perform operation 'Ingest' on 'any database'., ActivityId='<redacted>'
	at com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask.refreshIngestionResources(ResourceManager.java:170)
	at com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask.lambda$run$0(ResourceManager.java:136)
	at com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity.invoke(MonitoredActivity.java:33)
	at com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity.invoke(MonitoredActivity.java:26)
	at com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask.run(ResourceManager.java:135)
	at java.base\/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base\/java.util.TimerThread.run(Timer.java:516)
    Caused by: com.microsoft.azure.kusto.data.exceptions.DataServiceException: Principal 'aadapp=<app-id>;<tenant>' is not authorized to perform operation 'Ingest' on 'any database'., ActivityId='<redacted>'
        at com.microsoft.azure.kusto.data.http.HttpPostUtils.createExceptionFromResponse(HttpPostUtils.java:191)
        at com.microsoft.azure.kusto.data.http.HttpPostUtils.post(HttpPostUtils.java:72)
        at com.microsoft.azure.kusto.data.ClientImpl.lambda$executeToJsonResult$1(ClientImpl.java:224)
        at com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity.invoke(MonitoredActivity.java:33)
        at com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity.invoke(MonitoredActivity.java:26)
        at com.microsoft.azure.kusto.data.ClientImpl.executeToJsonResult(ClientImpl.java:223)
        at com.microsoft.azure.kusto.data.ClientImpl.executeImpl(ClientImpl.java:173)
        at com.microsoft.azure.kusto.data.ClientImpl.lambda$execute$0(ClientImpl.java:122)
        at com.microsoft.azure.kusto.data.instrumentation.MonitoredActivity.invoke(MonitoredActivity.java:33)
        at com.microsoft.azure.kusto.data.ClientImpl.execute(ClientImpl.java:121)
        at com.microsoft.azure.kusto.data.ClientImpl.execute(ClientImpl.java:116)
        at com.microsoft.azure.kusto.data.ClientImpl.execute(ClientImpl.java:111)
        at com.microsoft.azure.kusto.data.ClientImpl.execute(ClientImpl.java:106)
        at com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask.lambda$refreshIngestionResources$e961bae5$1(ResourceManager.java:153)
        at io.github.resilience4j.retry.Retry.lambda$decorateCheckedSupplier$3f69f149$1(Retry.java:137)
        at com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask.refreshIngestionResources(ResourceManager.java:154)
        ... 6 more
    Caused by: {
        \"error\": {
            \"code\": \"Forbidden\",
            \"message\": \"Caller is not authorized to perform this action\",
            \"@type\": \"Kusto.Common.Svc.Exceptions.UnauthorizedOperationException\",
            \"@message\": \"Principal 'aadapp=<redacted>;<redacted>' is not authorized to perform operation 'Ingest' on 'any database'.\",
            \"@context\": {
                \"timestamp\": \"2024-09-03T09:20:32.2913175Z\",
                \"serviceAlias\": \"INGEST-<redacted>\",
                \"machineName\": \"<redacted>\",
                \"processName\": \"Kusto.WinSvc.DM.Svc\",
                \"processId\": 5212,
                \"threadId\": 6624,
                \"clientRequestId\": \"KJC.execute;<redacted>\",
                \"activityId\": \"<redacted>\",
                \"subActivityId\": \"<redacted>\",
                \"activityType\": \"GW.ExecuteDmCommand\",
                \"parentActivityId\": \"<activity-id>\",
                \"activityStack\": \"(Activity stack: CRID=KJC.execute;<redacted> ARID=<activity-id> > GW.Http.CallContext\/<activity-id> > GW.ExecuteDmCommand\/<redacted>)\"
            },
            \"@permanent\": true
        }
    }
    at com.microsoft.azure.kusto.data.http.HttpPostUtils.createExceptionFromResponse(HttpPostUtils.java:174)
    ... 21 more",
    "exception_class":"com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException",
    "exception_message":"Error refreshing IngestionResources. Principal 'aadapp=<redacted>;<redacted>' is not authorized to perform operation 'Ingest' on 'any database'., ActivityId='<redacted>'"
  },
  "source_host":"connect-connect-dcb64659f-dl7w6",
  "method":"run",
  "level":"ERROR",
  "message":"Error in refreshIngestionResources: Error refreshing IngestionResources. Principal 'aadapp=<redacted>;<redacted>' is not authorized to perform operation 'Ingest' on 'any database'., ActivityId='<redacted>'",
  "mdc":{"connector.context":"[kusto-sink-connector-<redacted>task-3] "},
  "@timestamp":"2024-09-03T09:20:32.290Z",
  "file":"ResourceManager.java",
  "line_number":"140",
  "thread_name":"Timer-0",
  "@version":1,"
  logger_name":"com.microsoft.azure.kusto.ingest.ResourceManager",
  "class":"com.microsoft.azure.kusto.ingest.ResourceManager$RefreshIngestionResourcesTask"
}
```

Applying Ingestor role on DB level was not enough. Even Admin role was not enough on DB level.

Before using Pod Identity it was enough to have Ingestor role defined on Database level. Maybe there is a bug within the refresh resources operation. It should be database scoped, not "any database".

Setup:
- Strimzi Kafka Operator
- Kafka Connect Image: quay.io/strimzi/kafka:latest-kafka-3.8.0
- Kusto Sink Plugin: microsoftcorporation/kafka-sink-azure-kusto:4.1.1

**Comment by ag-ramachandran** [comment]
Hello @jrauschenbusch , thanks for reporting. Would you mind sending me the Kafka connector config and the cluster URL to my MS alias r a m a c g at m..soft and I can see what is happening in this case.

**Comment by jrauschenbusch** [comment]
Sure. Email or what?

**Comment by ag-ramachandran** [comment]
Yes sure, an email would be great

**Comment by ag-ramachandran** [comment]
Solved in PR #127

**Comment by jrauschenbusch** [comment]
When will the release be published on Confluent Hub?
